rust-test-program/src/main.rs

```diff
@@ -0,0 +1,209 @@
+#![no_std]
```

Comment:
Rather than using inline assembly in this file, we should be using regular rust code (likely with some unsafe logic) to achieve similar functionality. The main purpose of this program is to ensure that regularly compiled rust code works on the simulated cpu design.